### PR TITLE
Add metadata indexes to help with segment allocation.

### PR DESCRIPTION
**Pull request by gianm**
Segment allocation queries can take a long time (10s of seconds) when
you have a lot of segments. Adding these indexes helps greatly.

### CODE REVIEW

**Comment by QiuMM** [comment]
LGTM！

File: server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java [hunk 1]
```diff
@@ -259,8 +264,15 @@ public void createSegmentTable(final String tableName)
...
             ),
-            StringUtils.format("CREATE INDEX idx_%1$s_datasource ON %1$s(dataSource)", tableName),
-            StringUtils.format("CREATE INDEX idx_%1$s_used ON %1$s(used)", tableName)
+            StringUtils.format(
+                "CREATE INDEX idx_%1$s_datasource_end ON %1$s(dataSource, %2$send%2$s)",
+                tableName,
+                getQuoteString()
+            ),
+            StringUtils.format(
+                "CREATE INDEX idx_%1$s_datasource_sequence ON %1$s(dataSource, sequence_name)",
```
Comment (QiuMM) [inline server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java]: @gianm sorry,  I did not review these code carefully. There are no `sequence_name ` of `segmentTable`, `pendingSegmentsTable` does...

File: server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java [hunk 2]
```diff
@@ -215,6 +215,11 @@ public void createPendingSegmentsTable(final String tableName)
                 + "  UNIQUE (sequence_name_prev_id_sha1)\n"
                 + ")",
                 tableName, getPayloadType(), getQuoteString()
+            ),
+            StringUtils.format(
+                "CREATE INDEX idx_%1$s_datasource_used_end ON %1$s(dataSource, used, %2$send%2$s)",
```
Comment (QiuMM) [inline server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java]: No `used` field of `pendingSegmentsTable `.

File: server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java (hunk 1 above)
Comment (gianm) [reply server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java]: Ouch, you're right. Thanks for the catch- I will fix it. I'm surprised the tests didn't catch it - I thought they ran these create table commands. I will look into that too.

Comment (gianm) [reply server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java]: Raised #6356 to fix this. The PR description also explains why the tests didn't catch this.

**Pull request merged**